src/applications/personalization/dashboard/utils/helpers.js

```diff
@@ -27,3 +30,29 @@ export const currency = amount => {
   });
   return formatter.format(parseFloat(amount));
 };
+
+const v1BaseUrl = `${environment.API_URL}/my_health/v1`;
+
+export const getFolderList = () => {
+  return apiRequest(
+    `${v1BaseUrl}/messaging/folders?page=1&per_page=999&useCache=false`,
+    {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    },
+  );
+};
+
+export const countUnreadMessages = folders => {
+  let unreadMessageCount = 0;
+  if (Array.isArray(folders?.data)) {
+    unreadMessageCount = folders.data.reduce((accumulator, currentFolder) => {
+      return accumulator + currentFolder.attributes?.unreadCount;
+    }, 0);
+  } else if (folders?.data?.attributes?.unreadCount > 0) {
+    unreadMessageCount = folders.data.attributes.unreadCount;
+  }
+
+  return unreadMessageCount;
+};
```

Comment:
Non-blocking, my alternative way of writing this without a mutating count variable, but I think what you have is fine.

```js
export const countUnreadMessages = folders => {
  if (Array.isArray(folders?.data)) {
    return  folders.data.reduce((accumulator, currentFolder) => {
      return accumulator + currentFolder.attributes?.unreadCount;
    }, 0);
  }

  if (folders?.data?.attributes?.unreadCount > 0) {
   return folders.data.attributes.unreadCount;
  }

  return 0;
};
```